src/Fleck/WebSocketConnection.cs

```diff
@@ -56,38 +57,38 @@ public WebSocketConnection(ISocket socket, Action<IWebSocketConnection> initiali
       get { return !_closing && !_closed && Socket.Connected; }
     }
 
-    public void Send(string message)
+    public Task Send(string message)
     {
-      Send(message, Handler.FrameText);
+      return Send(message, Handler.FrameText);
     }
 
-    public void Send(byte[] message)
+    public Task Send(byte[] message)
     {
-      Send(message, Handler.FrameBinary);
+        return Send(message, Handler.FrameBinary);
     }
 
-    public void SendPing(byte[] message)
+    public Task SendPing(byte[] message)
     {
-      Send(message, Handler.FramePing);
+        return Send(message, Handler.FramePing);
     }
 
-    public void SendPong(byte[] message)
+    public Task SendPong(byte[] message)
     {
-      Send(message, Handler.FramePong);
+        return Send(message, Handler.FramePong);
     }
 
-    private void Send<T>(T message, Func<T, byte[]> createFrame)
+    private Task Send<T>(T message, Func<T, byte[]> createFrame)
     {
       if (Handler == null)
         throw new InvalidOperationException("Cannot send before handshake");
 
       if (!IsAvailable) {
         FleckLog.Warn("Data sent while closing or after close. Ignoring.");
-        return;
+        return null;

```

Comment:
How should I do that?
The Task's Status property is read-only, so it's impossible to create a task only for the sake of returning it with a Faulted status.
Also, a task was not yet created at this point, so it's not necessary to handle it in the way you suggested, other than for keeping the consumer code clean.

The only way I can think to accomplish it is to create a task, throw exception in it's body, start it, and return it.
Is this what you had in mind?